### ISSUE TITLE
Support FastGit as alternative download url

### ISSUE THREAD

**Issue by gzzchh** [zdharma/zinit#473]
**Is your feature request related to a problem? Please describe.**
GitHub Raw has block or jamed in China. Using fastgit will accelerate it.

**Describe the solution you'd like**
using an env var to enable using fastgit as aleternative download url
or check the download status. when download from github raw fails, switch to fastgit as fallback.

**Additional context**
also cloud provide an env var for user to define their own mirror/alternative download urls

**Comment by pschmitt** [comment]
Interesting. How does fastgit work from a user perspective? Do you simply replace the github.com domain in the checkout url?

**Comment by gzzchh** [comment]
almost is.
but the domain has a little different for different github services
https://doc.fastgit.org/en-us/guide.html#the-usage-of-web

**Comment by pschmitt** [comment]
Thanks for googling it for me ;)
I'm genuinely curious about fastgit.

So. For cloning this here repo for example you'd need to:

```
git clone https://download.fastgit.org/zdharma/zinit.git
```

While doing that I noticed that I got redirected to github.com:

```bash
$ git clone https://download.fastgit.org/zdharma/zinit.git
Cloning into 'zinit'...
warning: redirecting to https://github.com/zdharma/zinit.git/ # <-- HERE
remote: Enumerating objects: 130, done.
remote: Counting objects: 100% (130/130), done.
remote: Compressing objects: 100% (58/58), done.
remote: Total 19905 (delta 81), reused 116 (delta 72), pack-reused 19775
Receiving objects: 100% (19905/19905), 10.52 MiB | 10.61 MiB/s, done.
Resolving deltas: 100% (12854/12854), done.
```

Does fastgit lookup the source IP and redirect if not originating from China?

**Comment by gzzchh** [comment]
![image](https://user-images.githubusercontent.com/17796830/105856926-b6fc2880-6024-11eb-80ef-8cea37519f7f.png)

maybe yes. this is a test from a hk vps.

**Comment by Whisht** [comment]
It will redriect if you are not belong to Chinese IP. 
``` 
git clone https://hub.fastgit.org/zdharma/zinit.git
Cloning into 'zinit'...
remote: Enumerating objects: 130, done.
remote: Counting objects: 100% (130/130), done.
remote: Compressing objects: 100% (58/58), done.
remote: Total 19905 (delta 81), reused 116 (delta 72), pack-reused 19775
Receiving objects: 100% (19905/19905), 10.52 MiB | 129.00 KiB/s, done.
Resolving deltas: 100% (12854/12854), done.
```

**Comment by pschmitt** [comment]
I wonder if you can get away with an `/etc/hosts` entry.

**Comment by Whisht** [comment]
Unfortunately, `hosts` doesn't always work. And its speed is much slower than proxy.

**Comment by pschmitt** [comment]
I think you misunderstood what I meant.
What I suggested was editing the `hosts` file so that github.com points to the fastgit IPs

**Comment by Whisht** [comment]
Well, this doen't work. After I alter the IP of `github.com` as `fastgit.org` IP, remote server will ask a `certification` when clone files. Server rejects direct IP access. 
```
git clone https://github.com/zdharma/zinit.git
Cloning into 'zinit'...
fatal: unable to access 'https://github.com/zdharma/zinit.git/': SSL: certificate subject name (isaacchen.cn) does not match target host name 'github.com'
fatal: unable to access 'https://github.com/zdharma/zinit.git/': gnutls_handshake() failed: Error in the pull function.
```

**Comment by pschmitt** [comment]
Yeah, you'd obviously also need to disable TLS cert validation. Not exactly ideal :-/

**Comment by Whisht** [comment]
Sadly, now seems the easiset way is to repalace the clone url.

**Comment by gzzchh** [comment]
reference to this document  https://zdharma.org/zinit/wiki/Private-Repositories/
we can use `from` ice

like this 

```zsh
zinit lucid wait'1' for \
    from"hub.fastgit.org" \
    matthieusb/zsh-sdkman
```

**Comment by Whisht** [comment]
Hey, guys. I found an elegant way to solve this. Just config the git url:
`
git config --global url."https://hub.fastgit.org/".insteadOf "https://github.com/"
`
A fast mirror you will get!

**Comment by younger-1** [comment]
> Hey, guys. I found an elegant way to solve this. Just config the git url:
> `git config --global url."https://hub.fastgit.org/".insteadOf "https://github.com/"`
> A fast mirror you will get!

But this method didn't work for this kind of zinit config
```
zinit snippet OMZ::plugins/git/git.plugin.zsh
zinit snippet PZT::modules/helper/init.zsh
```
